squad/core/notification.py

```diff
@@ -63,10 +63,9 @@ def summary(self):
     def recipients(self):
         emails = []
         for sub in self.project.subscriptions.all():
-            if sub.user and sub.user.email:
-                emails.append(sub.user.email)
-            else:
-                emails.append(sub.email)
+            email = sub.get_email()
+            if email:
```

Comment:
email is garanteed to always be something given the model validation in place :-)

Reply:
Well, not really. We guarantee that user or email will be present, but what might happen is that user's email is not populated (if I recall it's not mandatory in django base User model).
Do you think we should do something about that (i.e. display the warning on subscription panel if user did not set up his email address and also do a separate validation on admin side)?

Reply:
yes, I guess such a warning would be useful.